src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -623,6 +626,11 @@ private bool TakeStartLine(SocketInput input)
                     return false;
                 }
 
+                char[] requestUriChars;
+                var rawPathLength = beginPath.GetCharArray(endPath, out requestUriChars);
+                var decodedLength = UrlPathDecoder.DecodeInPlace(requestUriChars, rawPathLength);
```

Comment:
To achieve the goal of in place updating in the memory pool block I need to significantly change the `UrlPathDecoder` so that it accepts two `iterator` like objects to operator. The existing logic heavily depends on random access for verify UTF8 coding.

I think it's do doable but we won't be able to share the `Decoder` since it will be highly specified for `Kestrel`. And I'm more worried about progress at this point to redo this decoder at this point. I'm more than willing to make it happen after rc1.